TunnelVisionLabs.ReferenceAssemblyAnnotator/SuppressibleLoggingHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace TunnelVisionLabs.ReferenceAssemblyAnnotator
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.Build.Utilities;
+
+    internal readonly struct SuppressibleLoggingHelper
+    {
+        private readonly ImmutableHashSet<string> _noWarn;
+
+        public SuppressibleLoggingHelper(TaskLoggingHelper helper, string? noWarn)
+        {
+            Helper = helper;
+
+            _noWarn = noWarn
+                ?.Split(new[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(item => item.Trim())
+                .Where(item => item.Length != 0)
+                .ToImmutableHashSet(StringComparer.OrdinalIgnoreCase) ?? ImmutableHashSet<string>.Empty;
```

Comment:
Suggestions welcome if you have other preferences!

Reply:
```suggestion
                .ToImmutableHashSet(StringComparer.Ordinal) ?? ImmutableHashSet<string>.Empty;
```

📝 You'll also be able to remove the `??` once the property is not nullable.

Reply:
Are all warning codes (CS*, NU*, IDE*, etc) case sensitive?

Reply:
My understanding is yes.

Reply:
I would be very sad if `ra1000` had a different meaning than `RA1000`, but consistency is good.